### DISCUSSION TITLE
Determine mouse position on the GraphEditor while regarding the displacement of the editor area

### DISCUSSION THREAD

**Discussion by TrickShotMLG02** [eckig/graph-editor#191] [Q&A] [answered]
I am trying to figure out on how to get the mouse position on the editor. However I also want to take the displacement of the editor itself into account, since it is possible to move the "drawing area" around and not start with 0,0 in the top left corner of the editor, but with something like 700,350. Is this possible in any way?

**Comment by eckig** [comment]
If I understand you correctly, this should work:
```
io.github.eckig.grapheditor.GraphEditor editor = ...;
editor.getView().addEventHandler(MouseEvent.MOUSE_MOVED, ev ->
{
  System.out.println(ev.getX()+", "+ ev.getY());
})
```

**Reply by TrickShotMLG02** [reply]
![image](https://github.com/eckig/graph-editor/assets/43967908/3737e1ee-2bfc-425f-854b-e017a2e5fa2c)
The top left corner is always (0,0). However I wanted it to be the abolute position on the graph itself instead of the position of the mouse on the panel. Thus if I have a node in the middle of the graph, I always get the exact position of the node on the graph and not relative to the current view

**Reply by eckig** [reply, marked as answer]
I got a typo in my snippet.. updated.
But if I use this snippet on the demo app, I get the absolute coordinates (the top left is only 0,0 if pan to the top-left-most position).
